### PR TITLE
add: Use opts.ResyncPeriod as controlPlaneInformerManager and sharedF…

### PR DESCRIPTION
**Pull request by wangxf1987**
…actory

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by RainbowMango** [comment]
Is this related to #4823 ?

**Comment by wangxf1987** [comment]
> Is this related to #4823 ?

not related

**Comment by wangxf1987** [comment]
@RainbowMango  I want to contribute our resource interperter, it name is fed.jd.com/v1/KService, which is equivalent to apps.kruise.io. It it possible?

**Comment by RainbowMango** [comment]
Do you mean to put `fed.jd.com/v1/KService` under [the thirdparty resourcecustomizations](https://github.com/karmada-io/karmada/tree/master/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations)? 

As far as I know, the third-party customizations are designed for those well-known open-sourced resource types. 
Is the `fed.jd.com/v1/KService` open-sourced?

**Comment by wangxf1987** [comment]
> Do you mean to put `fed.jd.com/v1/KService` under [the thirdparty resourcecustomizations](https://github.com/karmada-io/karmada/tree/master/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations)?
> 
> As far as I know, the third-party customizations are designed for those well-known open-sourced resource types. Is the `fed.jd.com/v1/KService` open-sourced?

open will be soon

**Comment by wangxf1987** [comment]
> Do you mean to put `fed.jd.com/v1/KService` under [the thirdparty resourcecustomizations](https://github.com/karmada-io/karmada/tree/master/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations)?
> 
> As far as I know, the third-party customizations are designed for those well-known open-sourced resource types. Is the `fed.jd.com/v1/KService` open-sourced?

we have an plan in the jdcloud, and open the kservice project

**Comment by RainbowMango** [comment]
> we have an plan in the jdcloud, and open the kservice project

Nice. Please talk to me then.

**Comment by wangxf1987** [comment]
> add the opts.ResyncPeriod in the controller-manager, I not found the NotSync issue. it's should be consider to accpet the PR?
@RainbowMango

File: pkg/resourceinterpreter/interpreter.go [hunk 1]
```diff
@@ -99,6 +99,7 @@ func (i *customResourceInterpreterImpl) Start(ctx context.Context) (err error) {
 	i.defaultInterpreter = native.NewDefaultInterpreter()
 
 	i.informer.Start()
+	i.informer.WaitForCacheSync()
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/interpreter.go:102]: This line does not seem to be directly related to the `ResyncPeriod`

**Comment by RainbowMango** [comment]
I'll look at it.
But I'm curious why you need this. Using resync to solve problems might not be a recommended pattern.

File: pkg/resourceinterpreter/interpreter.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/resourceinterpreter/interpreter.go:102]: So, what's your option?

I guess it's necessary, right?

**Comment by RainbowMango** [comment]
@wangxf1987 Xiaofei, this PR contains a merge commit that is not what we want. Could you please rebase your PR, you can find the guidelines at https://karmada.io/docs/next/contributor/github-workflow.

**Review by RainbowMango — APPROVED** [review]
/approve

Leave LGTM to @XiShanYongYe-Chang or @chaunceyjiang for final decision.

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/lgtm

**Pull request merged**